server/issue.go

```diff
@@ -307,6 +307,116 @@ func (s *Server) registerIssueRoutes(g *echo.Group) {
 		}
 		return nil
 	})
+
+	g.POST("/issue/:issueID/rollback", func(c echo.Context) error {
+		ctx := c.Request().Context()
+		issueID, err := strconv.Atoi(c.Param("issueID"))
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Issue ID %s is not a number", c.Param("issueID"))).SetInternal(err)
+		}
+		issue, err := s.store.GetIssueByID(ctx, issueID)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch issue with ID %d", issueID)).SetInternal(err)
+		}
+		if issue == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Issue ID not found: %d", issueID))
+		}
+
+		rollbackPayload := new(api.TaskRollbackRequestPayload)
+		if err := jsonapi.UnmarshalPayload(c.Request().Body, rollbackPayload); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed task rollback request").SetInternal(err)
+		}
+
+		if len(rollbackPayload.TaskIDList) != 1 {
+			return echo.NewHTTPError(http.StatusBadRequest, "The task ID list must have exactly one element")
+		}
+
+		taskID64, err := strconv.ParseInt(rollbackPayload.TaskIDList[0], 10, 64)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Invalid task ID %s", rollbackPayload.TaskIDList[0])
+		}
+		taskID := int(taskID64)
+		task, err := s.store.GetTaskByID(ctx, taskID)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to get task with ID %d", taskID)).SetInternal(err)
+		}
+		if task == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Task not found with ID %d", taskID))
+		}
+		if task.Type != api.TaskDatabaseDataUpdate {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Task type must be %s, but got %s", api.TaskDatabaseDataUpdate, task.Type))
+		}
+		if task.PipelineID != issue.PipelineID {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Task %d is not in pipeline %d", taskID, issue.PipelineID))
```

Comment:
pipeline concept is hidden from the user, let's use issue in the returned err msg instead